### PR TITLE
Bugfix/db init

### PR DESCRIPTION
**Pull request by jackcasstlesjones**
# What type of PR is this? (check all applicable)

# POTENTIALLY BREAKING CHANGE. PLEASE DO NOT MERGE UNTIL BOTH @JasonWarrenUK AND @maxitect HAVE REVIEWED.

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Styling
- [ ] Build
- [ ] Chore

## Description

- Wrapped db logic in a class to prevent errors from constant initialisation - previous `if` statement to check if db existed was not doing the job. Db was attempting to be initialised on page refresh and erroring.
- Had to comment out a lot of `addTool` with jason's permission

## Screenshots

_Please replace this line with any relevant images for UI changes._

## UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_

- [ ] Semantic HTML implemented?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

## Added/updated tests?

_Please aim to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [x] I need help with writing tests

## Delete branch after merge?

- [x] Yes
- [ ] No

## What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: src/app/insights/components/InsightsDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      {insights ? (
+        insights.map((insight, index) => (
+          <div key={index} className="mb-5">
+            <h1>dopamine: {insight.neurotransmitters.dopamine}</h1>
+            <h1>serotonin: {insight.neurotransmitters.serotonin}</h1>
+            <h1>adrenaline: {insight.neurotransmitters.adrenaline}</h1>
+          </div>
+        ))
+      ) : (
+        <h1>No insights available</h1>
```
Comment (maxitect) [inline src/app/insights/components/InsightsDisplay.tsx:42]:
```suggestion
            <p>dopamine: {insight.neurotransmitters.dopamine}</p>
            <p>serotonin: {insight.neurotransmitters.serotonin}</p>
            <p>adrenaline: {insight.neurotransmitters.adrenaline}</p>
          </div>
        ))
      ) : (
        <p>No insights available</p>
```
I know this is just for testing (and it works, I tested it in the deployed preview) but does not have to be offensively not accessible! xD

File: src/lib/db/DatabaseManager.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import { createRxDatabase, RxDatabase } from "rxdb";
+import toolkitItemSchema from "./schemas/toolkitItemSchema.json";
+import moodRecordSchema from "./schemas/moodRecordSchema.json";
+import categorySchema from "./schemas/categorySchema.json";
+import { v4 as uuidv4 } from "uuid";
+
+addRxPlugin(RxDBDevModePlugin);
+
+class DatabaseManager {
+  dbInstance: RxDatabase | null = null;
```
Comment (maxitect) [inline src/lib/db/DatabaseManager.ts:13]: I think this should be inside a constructor method if this is a class? And if not, why?

**Review by maxitect — CHANGES_REQUESTED** [review]
This is really neat, having all database methods within a class is a good move - OOP FTW!
Couple of comments, one I won't let go to do with accessibility (even if it is for testing purposes), the other a bit more open we can probably discuss tomorrow.
Other than that, I have reviewed all the code here, tested the functionality in the deployment preview, tested that it reads and writes to the collections in my local storage, everything works fine for me.

PS: unclear why so much of the add tool code had to be commented out but we can review that tomorrow

**Review by JasonWarrenUK — APPROVED** [review]

**Review by gurtatiLND — APPROVED** [review]

**Review by maxitect — APPROVED** [review]
Yay!

**Pull request merged**